### PR TITLE
feat: add zig pdk

### PR DESCRIPTION
**Pull request by usdogu**

### CODE REVIEW

**Comment by bhelx** [comment]
Looks great! I have a few questions but I'm going to let @nilslice review it and merge it

File: examples/basic.zig [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        plugin.setVar("a", "this is var a");
+    } else {
+        allocator.free(var_a_optional.?);
+    }
+    const var_a = plugin.getVar("a") catch unreachable orelse "";
+    defer allocator.free(var_a);
+
+    const thing = plugin.getConfig("thing") catch unreachable orelse "<unset by host>";
+
+    const output = std.fmt.allocPrint(allocator, "{{\"count\": {d}, \"config\": \"{s}\", \"a\": \"{s}\"}}", .{ count, thing, var_a }) catch unreachable;
```
Comment (bhelx) [inline examples/basic.zig:31]: Out of curiosity, does Zig have a json encoder or decoder implementation that can be used from Wasm?

Comment (usdogu) [reply examples/basic.zig:31]: Yes it does in std.json namespace, I just wrote it in plaintext for the sake of simplicity.

**Comment by nilslice** [comment]
Thank you @usdogu! This looks great. I see you are also still working on your branch, and I will watch the [Actions](https://github.com/usdogu/zig-pdk/actions/runs/3789037505) there before merging. It might be easier for us to collaborate if I merge this once your current action completes. Then we can run Actions on this repo as well. 

I will add you as a collaborator to this repo once I merge it in, so you can work on this code directly if you'd like.

**Pull request merged**